src/Stratis.Bitcoin.Features.Consensus/FullNodeBuilderConsensusExtension.cs

```diff
@@ -0,0 +1,179 @@
+                            .AddSingleton<IGetUnspentTransaction, ConsensusQuery>(provider => provider.GetService<ConsensusQuery>());
+                        new PosConsensusRulesRegistration().RegisterRules(fullNodeBuilder.Network.Consensus);
+                    });
+            });
+
+            return fullNodeBuilder;
+        }
+
+        public class PowConsensusRulesRegistration : IRuleRegistration
+        {
+            public void RegisterRules(IConsensus consensus)
+            {
+                consensus.HeaderValidationRules = new List<IHeaderValidationConsensusRule>()
+                {
+                    new HeaderTimeChecksRule(),
+                    new CheckDifficultyPowRule(),
+                    new BitcoinActivationRule(),
+                    new BitcoinHeaderVersionRule()
+                };
+
+                consensus.IntegrityValidationRules = new List<IIntegrityValidationConsensusRule>()
+                {
+                    new BlockMerkleRootRule()
+                };
+
+                consensus.PartialValidationRules = new List<IPartialValidationConsensusRule>()
+                {
+                    new SetActivationDeploymentsPartialValidationRule(),
+
+                    new TransactionLocktimeActivationRule(), // implements BIP113
+                    new CoinbaseHeightActivationRule(), // implements BIP34
+                    new WitnessCommitmentsRule(), // BIP141, BIP144
+                    new BlockSizeRule(),
+
+                    // rules that are inside the method CheckBlock
+                    new EnsureCoinbaseRule(),
+                    new CheckPowTransactionRule(),
+                    new CheckSigOpsRule(),
+                };
+
+                consensus.FullValidationRules = new List<IFullValidationConsensusRule>()
+                {
+                    new SetActivationDeploymentsFullValidationRule(),
+
+                    // rules that require the store to be loaded (coinview)
+                    new LoadCoinviewRule(),
+                    new TransactionDuplicationActivationRule(), // implements BIP30
+                    new PowCoinviewRule(), // implements BIP68, MaxSigOps and BlockReward calculation
+                    new SaveCoinviewRule()
+                };
+            }
+        }
+
+        public class PosConsensusRulesRegistration : IRuleRegistration
+        {
+            public void RegisterRules(IConsensus consensus)
+            {
+                consensus.HeaderValidationRules = new List<IHeaderValidationConsensusRule>()
+                {
+                    new HeaderTimeChecksRule(),
+                    new HeaderTimeChecksPosRule(),
+                    new StratisBugFixPosFutureDriftRule(),
+                    new CheckDifficultyPosRule(),
+                    new StratisHeaderVersionRule(),
+                    new ProvenHeaderCoinstakeRule(),
```

Comment:
According to the documentation, the size rule should come before coinstake rule.
amirite @dangershony @noescape00 ?